### PR TITLE
split data-navigator-tap

### PR DESCRIPTION
**Pull request by a13**
I hope this one won't break your own workflow.

### CODE REVIEW

File: data-navigator-tap.el [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; run M-x tap-socket-start
```
Comment (hkjels) [inline data-navigator-tap.el]: I presume this should be changed to `data-navigator-tap-mode`.

File: data-navigator-tap.el [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+(defcustom data-navigator-tap-server-port
+  nil
+  "The port number that the tap WebSocket server should listen on."
+  :type 'number
+  :group 'data-navigator-tap)
+
+(defun data-navigator-tap--handle-incoming-data (_websocket frame)
+  "Handle incoming FRAME."
+  (let ((data (websocket-frame-text frame)))
+    (data-navigator-from-input data)))
```
Comment (hkjels) [inline data-navigator-tap.el:48]: We could send "tap>" as the second parameter here?  I'm thinking about making the Clojure side of this a library as well. At that point, we could even send the filename and line-number perhaps

**Comment by hkjels** [comment]
Looks really good. With the changes mentioned, I think this could be merged.
I would however like to research if integrating with Cider is doable as I think that would have some added benefit. https://github.com/hkjels/data-navigator.el/issues/7

**Comment by a13** [comment]
Sorry, what changes are you talking about?

**Comment by hkjels** [comment]
I made some review comments

**Review by hkjels — CHANGES_REQUESTED** [review]

File: data-navigator-tap.el (hunk 2 above)
Comment (a13) [reply data-navigator-tap.el:48]: I'll take a look

File: data-navigator-tap.el (hunk 1 above)
Comment (a13) [reply data-navigator-tap.el]: fixed, thanks

File: data-navigator-tap.el (hunk 2 above)
Comment (a13) [reply data-navigator-tap.el:48]: > We could send "tap>" as the second parameter here?

@hkjels can we postpone this one for now? 

> I'm thinking about making the Clojure side of this a library as well. 

I agree, but I think we need to discuss the best way to implement it (together with Cider integration etc)

**Pull request merged**